cpp/tests/table/table_view_tests.cu

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright (c) 2019, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <cudf/column/column_view.hpp>
+#include <cudf/table/table_view.hpp>
+#include <cudf/table/table_device_view.cuh>
+#include <cudf/table/table_device_view.cuh>
+#include <tests/utilities/base_fixture.hpp>
+#include <tests/utilities/type_lists.hpp>
+#include <tests/utilities/column_wrapper.hpp>
+#include <cudf/table/row_operators.cuh>
+#include <tests/utilities/column_utilities.hpp>
+
+#include <vector>
+#include <gmock/gmock.h>
+
+// Compares two tables row by row, if table1 row is less than table2, then corresponding row value in `ouput` would be `true`/1 else `false`/0.
+struct TableViewTest : public cudf::test::BaseFixture {};
+void row_comparison(cudf::table_view input1, cudf::table_view input2, cudf::mutable_column_view output, std::vector<cudf::order> const& column_order) {
+    cudaStream_t stream = 0;
+
+    auto device_table_1 = cudf::table_device_view::create(input1, stream);
+    auto device_table_2 = cudf::table_device_view::create(input2, stream);
+    rmm::device_vector<cudf::order> d_column_order(column_order);
+
+    auto comparator = cudf::experimental::row_lexicographic_comparator<false>( *device_table_1, *device_table_2,
+                                                           cudf::null_order::AFTER, d_column_order.data().get());
+
+    thrust::transform(rmm::exec_policy(stream)->on(stream),
+                     thrust::make_counting_iterator(0), thrust::make_counting_iterator(input1.num_rows()), thrust::make_counting_iterator(0), output.data<int8_t>(), comparator);
+
+}
+
+TEST_F(TableViewTest, EmptyColumnedTable)
+{
+    std::vector<cudf::column_view> cols{};
+
+    cudf::table_view input(cols);
+    cudf::size_type expected = 0;
+
+    EXPECT_EQ(input.num_columns(), expected);
+}
+
+TEST_F(TableViewTest, TestLexiologicalComparatorTwoTableCase)
```

Comment:
Lexiological? Do you mean `Lexicographical`?

Reply:
My bad